system-int-test/test/cmr/system_int_test/search/generic_association_test.clj

```diff
@@ -45,6 +45,216 @@
+                       token tl2-concept-id tl2-revision-id [{:concept-id tl1-concept-id  :revision-id tl1-revision-id}])
+            ;;Switch the position of tl1 and tl2 should return the same concept-id and revision-id is increased by 1.
+            response2 (association-util/generic-associate-by-concept-ids-revision-ids
+                       token tl1-concept-id tl1-revision-id [{:concept-id tl2-concept-id :revision-id tl2-revision-id}])
+            ;;Try to associate tl1 with tl2 by concept-id only. This shouldn't be allowed.
+            ;;Try to associate tl2 with tl1 by conept-id only,  This shouldn't be allowed.
+            ;;The following association is trying to do both of the above.
+            response3 (association-util/generic-associate-by-concept-ids-revision-ids
+                       token tl2-concept-id nil [{:concept-id tl1-concept-id}])
+            _ (index/wait-until-indexed)
+
+            ;;Search for the tool tl1, it should return the association
+            tl1-search-result (search-request "tools.umm_json" "native_id=tl1")
+            tl1-search-body (json/parse-string (:body tl1-search-result) true)
+            tl1-search-generic-associations (get-in (first (:items tl1-search-body)) [:meta :associations :tools])
+
+            ;;Search for the tool tl2, it should return the association
+            tl2-search-result (search-request "tools.umm_json" "native_id=tl2")
+            tl2-search-body (json/parse-string (:body tl2-search-result) true)
+            tl2-search-generic-associations (get-in (first (:items tl2-search-body)) [:meta :associations :tools])
+
+            ;;Dissociate the association
+            response4 (association-util/generic-dissociate-by-concept-ids-revision-ids
+                       token tl1-concept-id tl1-revision-id [{:concept-id tl2-concept-id :revision-id tl2-revision-id}])
+            _ (index/wait-until-indexed)
+
+            ;;Search for the tool tl1 again, it should NOT return the association
+            tl1-search-result1 (search-request "tools.umm_json" "native_id=tl1")
+            tl1-search-body1 (json/parse-string (:body tl1-search-result1) true)
+            tl1-search-generic-associations1 (get-in (first (:items tl1-search-body1)) [:meta :associations :tools])
+
+            ;;Search for the tool tl2 again, it should NOT return the association
+            tl2-search-result1 (search-request "tools.umm_json" "native_id=tl2")
+            tl2-search-body1 (json/parse-string (:body tl2-search-result1) true)
+            tl2-search-generic-associations1 (get-in (first (:items tl2-search-body1)) [:meta :associations :tools])]
+        ;; The first and second associations are successful
+        (is (= 200 (:status response1) (:status response2)))
+
+        ;; The first and second associations are creating and updating the same association
+        (is (= (get-in (first (:body response1)) [:generic-association :concept-id])
```

Comment:
nit: You don't have to do this, but it would be better to put the association response verification and search result containing association verification in their separate functions and then you can call them in this test rather than repeating all these details multiple times.

Reply:
Will write a separate ticket to simplify it now I'm done with all the association tests.